src/request/plan.rs

```diff
@@ -63,42 +66,230 @@ impl<Req: KvRequest + HasKeys> HasKeys for Dispatch<Req> {
+                    // don't sleep if we have resolved the region error
+                    if !region_error_resolved {
+                        futures_timer::Delay::new(duration).await;
+                    }
+                    Self::single_plan_handler(pd_client, plan, backoff, permits).await
+                }
+                None => Err(e),
+            }
+        } else {
+            Ok(vec![Ok(resp)])
+        }
+    }
+
+    // Returns
+    // 1. Ok(true): error has been resolved, retry immediately
+    // 2. Ok(false): backoff, and then retry
+    // 3. Err(Error): can't be resolved, return the error to upper level
+    async fn handle_region_error(
+        pd_client: Arc<PdC>,
+        error: Error,
+        region_store: RegionStore,
+    ) -> Result<bool> {
+        match error {
+            Error::RegionError(mut e) => {
+                let ver_id = region_store.region_with_leader.ver_id();
+                if e.has_not_leader() {
+                    let not_leader = e.get_not_leader();
+                    if not_leader.has_leader() {
+                        match pd_client
+                            .update_leader(
+                                region_store.region_with_leader.ver_id(),
+                                not_leader.get_leader().clone(),
+                            )
+                            .await
+                        {
+                            Ok(_) => Ok(true),
+                            Err(Error::EntryNotFoundInRegionCache) => Ok(false),
+                            Err(e) => {
+                                pd_client.invalidate_region_cache(ver_id).await;
+                                Err(e)
+                            }
+                        }
+                    } else {
+                        // The peer doesn't know who is the current leader. Generally it's because
+                        // the Raft group is in an election, but it's possible that the peer is
+                        // isolated and removed from the Raft group. So it's necessary to reload
+                        // the region from PD.
+                        pd_client.invalidate_region_cache(ver_id).await;
+                        Ok(false)
+                    }
+                } else if e.has_store_not_match() {
+                    pd_client.invalidate_region_cache(ver_id).await;
+                    Ok(false)
+                } else if e.has_epoch_not_match() {
+                    Self::on_region_epoch_not_match(
+                        pd_client.clone(),
+                        region_store,
+                        e.take_epoch_not_match(),
+                    )
+                    .await
+                } else if e.has_server_is_busy()
+                    || e.has_stale_command()
+                    || e.has_region_not_found()
+                    || e.has_max_timestamp_not_synced()
+                {
+                    pd_client.invalidate_region_cache(ver_id).await;
```

Comment:
I think `ServerIsBusy` and `MaxTimestampNotSynced` don't need to invalidate cache.